client/my-sites/importer/importing-pane.jsx

```diff
@@ -193,7 +194,7 @@ class ImportingPane extends React.PureComponent {
 	}
 
 	handleOnMap = ( source, target ) =>
-		defer( () => mapAuthor( get( this.props, 'importerStatus.importerId' ), source, target ) );
+		this.props.mapAuthor( this.props.importerStatus.importerId, source, target );
```

Comment:
fun fact in case it's not obvious. when this code was born we didn't use Redux in Calypso IIRC. this meant we didn't have synchronous updates. calling `mapAuthor` directly and other mutative/changing calls would lead to render issues.

glad to see you put it to rest.

Reply:
Thanks @dmsnell for the additional info. Typically, `defer` is needed to avoid nested Flux dispatches. Flux doesn't like when an action is dispatched while another dispatch is still running. That can easily happen in a React render cycle:
```
dispatch( a ) -> store change event -> setState -> rerender -> componentDidUpdate -> dispatch( b )
```
When all that happens synchronously within one call stack (as it often does), exceptions are thrown from Flux.

Redux doesn't have that issue: the state update with reducers is atomic, and even with nested dispatches (dispatch -> subscription handler -> another dispatch) the state progression state1 -> state2 -> state3 is well defined.

Reply:
exactly that! 😄